src/Nethermind/Nethermind.Synchronization/BeamSync/BeamBlockchainProcessor.cs

```diff
@@ -260,18 +261,26 @@ private void BeamProcess(Block block)
                 _recoveryStep.RecoverData(block);
                 (IBlockchainProcessor beamProcessor, IStateReader stateReader) = CreateProcessor(block, new ReadOnlyDbProvider(_readOnlyDbProvider, true), _specProvider, _logManager);
 
-                BlockHeader parentHeader = _readOnlyBlockTree.FindHeader(block.ParentHash, BlockTreeLookupOptions.TotalDifficultyNotNeeded);
+                BlockHeader? parentHeader = _readOnlyBlockTree.FindHeader(block.ParentHash, BlockTreeLookupOptions.TotalDifficultyNotNeeded);
                 if (parentHeader != null)
                 {
-                    prefetchTasks = PrefetchNew(stateReader, block, parentHeader.StateRoot, parentHeader.Author ?? parentHeader.Beneficiary);
+                    if (parentHeader.StateRoot == null)
+                    {
+                        throw new InvalidDataException($"Received a block with null {nameof(parentHeader.StateRoot)} for beam processing");
+                    }
+                    if (parentHeader.GasBeneficiary == null)
+                    {
+                        throw new InvalidDataException($"Received a block with null {nameof(parentHeader.GasBeneficiary)} for beam processing");
+                    }
+                    prefetchTasks = PrefetchNew(stateReader, block, parentHeader.StateRoot, parentHeader.GasBeneficiary);
```

Comment:
`GasBeneficiary` **is** `Author ?? Beneficiary`, so it felt appropriate to just use that rather than putting `Author ?? Beneficiary` in here.  Maybe there is a gotcha that makes this a bad idea?

Reply:
it may be old code from before the property was added